src/Chapter2.hs

```diff
@@ -842,7 +854,8 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [Int] -> [Int]
+rotate n l = if (n >= 0) then take (length l) (drop n (cycle l)) else []
```

Comment:
That's a correct solution! Though, it does a little bit more work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: